src/OrchardCore.Modules/OrchardCore.Users/TimeZone/Services/UserTimeZoneService.cs

```diff
@@ -6,82 +6,84 @@
-            _clock = clock;
-            _distributedCache = distributedCache;
-            _httpContextAccessor = httpContextAccessor;
-            _userManager = userManager;
-        }
+        _clock = clock;
+        _distributedCache = distributedCache;
+        _httpContextAccessor = httpContextAccessor;
+        _userManager = userManager;
+    }
 
-        public async Task<ITimeZone> GetUserTimeZoneAsync()
-        {
-            var currentTimeZoneId = await GetCurrentUserTimeZoneIdAsync();
-            if (string.IsNullOrEmpty(currentTimeZoneId))
-            {
-                return null;
-            }
+    public async Task<ITimeZone> GetUserTimeZoneAsync()
+    {
+        var currentTimeZoneId = await GetCurrentUserTimeZoneIdAsync();
 
-            return _clock.GetTimeZone(currentTimeZoneId);
+        if (string.IsNullOrEmpty(currentTimeZoneId))
+        {
+            return null;
         }
 
-        public Task UpdateUserTimeZoneAsync(User user)
-        {
-            var userName = user?.UserName;
+        return _clock.GetTimeZone(currentTimeZoneId);
+    }
 
-            if (!string.IsNullOrEmpty(userName))
-            {
-                return _distributedCache.RemoveAsync(GetCacheKey(userName));
-            }
+    public async Task UpdateUserTimeZoneAsync(IUser user)
+    {
+        var userName = user?.UserName;
 
-            return Task.CompletedTask;
+        if (!string.IsNullOrEmpty(userName))
+        {
+            await _distributedCache.RemoveAsync(GetCacheKey(userName));
         }
 
-        public async Task<string> GetCurrentUserTimeZoneIdAsync()
-        {
-            var userName = _httpContextAccessor.HttpContext.User?.Identity?.Name;
+        return;
+    }
 
-            if (string.IsNullOrEmpty(userName))
-            {
-                return null;
-            }
+    public async Task<string> GetCurrentUserTimeZoneIdAsync()
+    {
+        var userName = _httpContextAccessor.HttpContext.User?.Identity?.Name;
 
-            var key = GetCacheKey(userName);
-            var timeZoneId = await _distributedCache.GetStringAsync(key);
+        if (string.IsNullOrEmpty(userName))
+        {
+            return null;
+        }
 
-            if (string.IsNullOrEmpty(timeZoneId))
-            {
-                var user = await _userManager.FindByNameAsync(userName) as User;
-                timeZoneId = user.As<UserTimeZone>()?.TimeZoneId;
+        var key = GetCacheKey(userName);
+        var timeZoneId = await _distributedCache.GetStringAsync(key);
 
-                if (!string.IsNullOrEmpty(timeZoneId))
-                {
-                    await _distributedCache.SetStringAsync(key, timeZoneId, new DistributedCacheEntryOptions { SlidingExpiration = _slidingExpiration });
-                }
-            }
+        if (timeZoneId is null)
+        {
+            var user = await _userManager.FindByNameAsync(userName) as User;
+            timeZoneId = user.As<UserTimeZone>()?.TimeZoneId;
```

Comment:
user can also be null here (experience again(

Reply:
fixed